react/src/components/PrometheusMetricGraph.tsx

```diff
@@ -0,0 +1,255 @@
+import {
+  convertBinarySizeUnit,
+  toFixedFloorWithoutTrailingZeros,
+} from '../helper';
+import { useCurrentUserInfo } from '../hooks/backendai';
+import BAICard from './BAICard';
+import {
+  PrometheusMetricGraphQuery,
+  PrometheusMetricGraphQuery$data,
+} from './__generated__/PrometheusMetricGraphQuery.graphql';
+import { Empty, theme } from 'antd';
+import { createStyles } from 'antd-style';
+import graphql from 'babel-plugin-relay/macro';
+import dayjs from 'dayjs';
+import _ from 'lodash';
+import { useLazyLoadQuery } from 'react-relay';
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+  ReferenceLine,
+} from 'recharts';
+
+const useStyle = createStyles(({ css, token }) => ({
+  recharts: css`
+    .recharts-label {
+      fill: ${token.colorTextDescription};
+    }
+    .recharts-cartesian-axis-line {
+      stroke: ${token.colorBorder};
+    }
+    .recharts-cartesian-axis-tick-line {
+      stroke: ${token.colorBorder};
+    }
+    .recharts-cartesian-axis-tick-value {
+      fill: ${token.colorTextDescription};
+    }
+  `,
+}));
+
+type MetricData = NonNullable<
+  NonNullable<
+    PrometheusMetricGraphQuery$data['user_utilization_metric']
+  >['metrics']
+>;
+
+interface PrometheusMetricGraphProps {
+  startDate: string;
+  endDate: string;
+  metricName: string;
+  fetchKey: string;
+}
+
+const PrometheusMetricGraph: React.FC<PrometheusMetricGraphProps> = ({
+  startDate,
+  endDate,
+  metricName,
+  fetchKey,
+}) => {
+  const { token } = theme.useToken();
+  const { styles } = useStyle();
+  const userInfo = useCurrentUserInfo();
+  const dayDiff = dayjs(Number(endDate) * 1000).diff(
+    dayjs(Number(startDate) * 1000),
+    'day',
+  );
+
+  const { user_utilization_metric } =
```

Comment:
Instead of querying this component, why don't you use fragment?

Reply:
The queries used by the `PrometheusMetric` component and the `PrometheusMetricGraph` component are different queries.

It receives the metrics available through `container_utilization_metric_metadata` and passes information about each metric to `user_utilization_metric`.

Reply:
yes but i think it's better to handle all the queries in parent component and send fragments to children components.

Reply:
The metric_name input type of `user_utilization_metric` is a string type, so only one metric can be retrieved at a time.

Can I perform multiple queries in parallel in relay?

Reply:
Of course. Please refer LegacyFolderExplorer or DashboardPage.

Reply:
We decided to keep the existing form because it's not about sending separate queries within a single request, but about calling the same query multiple times with only the input changing.